fixes #28 where making time struct with New() function did not counted leap year logic

abtinokhovat opened the pull request:
New(time) function was not calculating correctly on days which were leap days or in leap years

`tl := time.Date(2025, 3, 20, 0, 0, 0, 0, time.Local)
	ti2 := New(tl)
	fmt.Println(ti2.Format("yyyy-MM-dd"))

	ti := Date(1403, Esfand, 28, 0, 0, 0, 0, time.Local)
	ti = ti.AddDate(0, 0, 2)

	fmt.Println(ti.Format("yyyy-MM-dd"))
`

for example these days are the same how ever the calculation had errors both have to be `1403-12-30` but the one with `ptime` package is `1404-01-01`

also this fixes this [issue](https://github.com/yaa110/go-persian-calendar/issues/27).

yaa110 commented:
@abtinokhovat Thank you for fixing the issue. Could you please add some unit tests for it as well?

seyedmmousavi commented:
It is a good idea to document the calculation reference. Based on which method and mathematical formula have you done the calculations?

abtinokhovat commented:
I'm now adding some documentation to the mathematic formula's I've used here and I will also add some tests here.

abtinokhovat commented:
@seyedmmousavi @yaa110 I've added documentation as best as I could. The formula is somewhat ambiguous, and the result is based on my work on calendars from various sources over the years. I've included them as I remembered and had them. It is legacy code that I originally wrote in TypeScript and then rewrote in Go. I also added detailed test cases for each conversion function, both forward and backward.

yaa110 commented:
fixes #28
